docs/non-normative/code-generation.md

```diff
@@ -0,0 +1,187 @@
+- Namespace: `opentelemetry.semconv` and `opentelemetry.semconv.incubating`
+- All supported Schema URLs should be listed to allow different instrumentations in the same application to provide the exact version of conventions they follow.
+- Attributes, metrics, and other convention definitions should be grouped by the convention type and the root namespace. See the example below:
+
+```
+├── SchemaUrls.code
+├── attributes
+│   ├── ClientAttributes.code
+│   ├── HttpAttributes.code
+│   └── ...
+├── metrics
+│   ├── HttpMetrics.code
+│   └── ...
+└── events
+    └── ...
+```
+
+## Generating semantic conventions
+
+This section describes how to do code-generation with weaver.
+
+> Note:
+> We're transitioning from [build-tools](https://github.com/open-telemetry/build-tools/blob/main/semantic-conventions/README.md#code-generator)
+> to [opentelemetry-weaver](https://github.com/open-telemetry/weaver/blob/main/crates/weaver_forge/README.md) to generate code for semantic conventions.
+> All new code-generation should be done using weaver, build-tools may become incompatible with future version of semantic conventions.
+
+Code-generation is based on YAML definitions in the specific version of semantic conventions and usually
+involves several steps which could be semi-automated:
+
+1. Manually update the Semantic Conventions version in config
+2. Add the new Schema URL to the list of supported versions
+   - If it's not automated, then it can, at least, be automatically checked.
+3. Check out (or download) the new version of Semantic Conventions
+4. Run code-generation script (see below for the details)
+5. Fix lint violations in the auto-generated code (if any)
+6. Send the PR with new code to the corresponding repository
+
+Here are examples of how steps 2-5 are implemented for [Python](https://github.com/open-telemetry/opentelemetry-python/pull/4091) and [Erlang](https://github.com/open-telemetry/opentelemetry-erlang/pull/733).
+
+Step 4 (running code generation) depends on language-specific customizations. It's also the only step that's affected by tooling migration.
+
+Check out[weaver code-generation documentation for the details](https://github.com/open-telemetry/weaver/blob/main/crates/weaver_forge/README.md)
+
+### Migrating from build-tools
+
+Migration from build-tools involves changing Jinja templates and adding [weaver config file](https://github.com/open-telemetry/weaver/blob/main/crates/weaver_forge/README.md#configuration-file---weaveryaml).
+
+#### Weaver config
+
+Here's a simplified example of this file that generates all attributes.
+
+```yaml
+params:
+  excluded_namespaces: [ios, aspnetcore, signalr, android, dotnet, jvm, kestrel]
+
+templates:
+  - pattern: semantic_attributes.j2
+    filter: >
+      semconv_grouped_attributes({
+        "exclude_root_namespace": $excluded_namespaces
+      })
+      | map({
+          root_namespace: .root_namespace,
+          attributes: .attributes,
+          output: $output
+        })
```

Comment:
Do we really need this with the new filters Laurent provided? In my .NET current WIP Code, I get all of this already with this simple weaver config file:

```
templates:
  - pattern: SemanticConventionsAttributes.cs.j2
    filter: >
      semconv_grouped_attributes
    application_mode: each
```

Reply:
I agree. I'm not sure that most of the codegen authors will need the output field, which is specific to the Python codegen.

Reply:
my goal was to demonstrate that you can  post-process and add more things into the context.

The reason python does it is because it generates metrics in addition to attributes and needs an extra folder structure for it. I think everyone would need some form of it.

I was a bad influence and now erlang does it too https://github.com/open-telemetry/opentelemetry-erlang/pull/733/files#diff-1a4c4f5071244fadabb10f3aed3958e7e4486869f3260201dc70509330fdc082

Reply:
In that case, why not directly use the `$output` parameter from the template (i.e. ctx.params.output)? I don't see the value in including it from a JQ projection.

Reply:
I updated the example to be more realistic and to modify the output `output: $output + "attributes/"`